### PR TITLE
Feature/qm share extension

### PR DESCRIPTION
**Pull request by VitaliyGuru**

### CODE REVIEW

File: QMShareExtension/Data Source/QMShareDataSource/QMShareDataSource.m [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    
+    if (self.showContactsSection) {
+        
+        if (indexPath.section == kContactsSection) {
+            
+            QMShareContactsTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:[QMShareContactsTableViewCell cellIdentifier]
+                                                                                 forIndexPath:indexPath];
+            
+            [self.contactsDataSource.selectedItems removeAllObjects];
```
Comment (Pro100Andrey) [inline QMShareExtension/Data Source/QMShareDataSource/QMShareDataSource.m]: Need to discuss

File: QMShareExtension/Data Source/QMShareItemsDataProvider/QMShareItemsDataProvider.m [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+//  Copyright © 2017 Quickblox. All rights reserved.
+//
+
+#import "QMShareItemsDataProvider.h"
+#import "QMSearchProtocols.h"
+
+@interface QMShareItemsDataProvider()
+
+@property (copy, nonatomic) NSString *cachedSearchText;
+@property (strong, nonatomic) NSArray *shareItems;
```
Comment (Pro100Andrey) [inline QMShareExtension/Data Source/QMShareItemsDataProvider/QMShareItemsDataProvider.m]: use generics

File: QMShareExtension/Data Source/QMShareItemsDataProvider/QMShareItemsDataProvider.m [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    
+    if (searchText.length == 0) {
+        
+        [self.dataSource replaceItems:self.shareItems];
+        [self.delegate searchDataProviderDidFinishDataFetching:self];
+        
+        return;
+    }
+
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
```
Comment (Pro100Andrey) [inline QMShareExtension/Data Source/QMShareItemsDataProvider/QMShareItemsDataProvider.m]: need check

File: QMShareExtension/View Controllers/QMShareTableViewController.h [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+- (void)didTapShareBarButtonWithSelectedItems:(NSArray<id<QMShareItemProtocol>> *)selectedItems;
+- (void)didTapCancelBarButton;
+- (void)didCancelSharing;
+
+@end
+
+@interface QMShareTableViewController : UITableViewController
+
++ (instancetype)qm_shareTableViewControllerWithDialogs:(NSArray *)dialogs
```
Comment (Pro100Andrey) [inline QMShareExtension/View Controllers/QMShareTableViewController.h]: generics

File: Q-municate/Categories/QBSettings/QBSettings+Qmunicate.m [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+static NSString * const kQMAccountKey = @"6Qyiz3pZfNsex1Enqnp7";
+
+#endif
+
+@implementation QBSettings (Qmunicate)
+
+
++ (void)setQmunicateSettings {
+    // Quickblox settings
+    [QBSettings setApplicationID:kQMApplicationID];
```
Comment (Pro100Andrey) [inline Q-municate/Categories/QBSettings/QBSettings+Qmunicate.m]: class property

File: Q-municate/Classes/Common/QMActivityItem/QMActivityItem.m [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    return item;
+}
+
+- (NSString *)activityViewController:(UIActivityViewController *)__unused activityViewController
+   dataTypeIdentifierForActivityType:(NSString *)__unused activityType {
+    return self.typeIdentifier;
+}
+
+static NSString *typeIdentifierForActivityItem(id item) {
+    if ([item isKindOfClass:[NSURL class]]) {
```
Comment (Pro100Andrey) [inline Q-municate/Classes/Common/QMActivityItem/QMActivityItem.m]: use once @{(id)QBSettings.class:@"class"};

File: Q-municate/Classes/Common/QMExtensionCache/QMExtensionCache.m [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    dispatch_once(&onceToken, ^{
+        [QMUsersCache setupDBWithStoreNamed:kQMUsersCacheNameKey
+                 applicationGroupIdentifier:[self appGroupIdentifier]];
+        usersCache =  QMUsersCache.instance;
+    });
+    
+    return usersCache;
+}
+
+@end
```
Comment (Pro100Andrey) [inline Q-municate/Classes/Common/QMExtensionCache/QMExtensionCache.m]: Q-municate/Classes/QMTasks/QMTasks.h <- ...icate/Classes/QMProfile/QMTasks/QMTasks.h

File: Q-municate/Classes/Tools/QMMediaController/QMMediaController.m [hunk 8]
```diff
@@ -731,5 +733,19 @@ - (UIView *)photosViewController:(NYTPhotosViewController *)__unused photosViewC
...
+//MARK: - QMAttachmentContentServiceDelegate
+- (BOOL)attachmentContentService:(QMAttachmentContentService *)__unused contentService
+        shouldDownloadAttachment:(QBChatAttachment *)attachment
+                       messageID:(NSString *)__unused messageID {
+    
+    return attachment.attachmentType == QMAttachmentContentTypeImage ||
+    attachment.attachmentType == QMAttachmentContentTypeAudio;
+}
+
+
```
Comment (Pro100Andrey) [inline Q-municate/Classes/Tools/QMMediaController/QMMediaController.m]: remove /n/n/n

File: Q-municate/Categories/QBSettings/QBSettings+Qmunicate.m (hunk 5 above)
Comment (VitaliyGuru) [reply Q-municate/Categories/QBSettings/QBSettings+Qmunicate.m]: done

**Pull request merged**